crates/e2e/src/setup/onchain_components/safe.rs

```diff
@@ -1,105 +1,133 @@
 use {
     super::{OnchainComponents, TestAccount},
-    contracts::{
-        GnosisSafe,
+    crate::nodes::NODE_HOST,
+    alloy::{
+        primitives::{Address, Bytes, U256},
+        providers::Provider,
+    },
+    contracts::alloy::{
+        GnosisSafe::{self, GnosisSafe::execTransactionCall},
         GnosisSafeCompatibilityFallbackHandler,
         GnosisSafeProxy,
         GnosisSafeProxyFactory,
     },
-    ethcontract::{Bytes, H160, H256, U256},
+    ethcontract::transaction::TransactionBuilder,
+    ethrpc::{
+        AlloyProvider,
+        alloy::{
+            ProviderSignerExt,
+            conversions::{IntoAlloy, TryIntoAlloyAsync},
+            provider,
+        },
+    },
     hex_literal::hex,
     model::{
         DomainSeparator,
         order::OrderCreation,
         signature::{Signature, hashed_eip712_message},
     },
-    shared::ethrpc::Web3,
-    web3::{
-        signing,
-        signing::{Key, SecretKeyRef},
-    },
+    std::marker::PhantomData,
+    web3::signing::{self},
 };
 
-macro_rules! tx_safe {
-    ($acc:expr_2021, $safe:ident, $call:expr_2021) => {{
-        let call = $call;
-        $crate::tx!(
-            $acc,
-            $safe.exec_transaction(
-                call.tx.to.unwrap(),
-                call.tx.value.unwrap_or_default(),
-                ::ethcontract::Bytes(call.tx.data.unwrap_or_default().0),
-                Default::default(),
-                Default::default(),
-                Default::default(),
-                Default::default(),
-                Default::default(),
-                Default::default(),
-                $crate::setup::safe::gnosis_safe_prevalidated_signature($acc.address()),
-            )
-        );
-    }};
-}
-
 pub struct Infrastructure {
-    pub factory: GnosisSafeProxyFactory,
-    pub fallback: GnosisSafeCompatibilityFallbackHandler,
-    pub singleton: GnosisSafe,
-    web3: Web3,
+    pub factory: GnosisSafeProxyFactory::Instance,
+    pub fallback: GnosisSafeCompatibilityFallbackHandler::Instance,
+    pub singleton: GnosisSafe::Instance,
+
+    pub provider: AlloyProvider,
 }
 
 impl Infrastructure {
-    pub async fn new(web3: &Web3) -> Self {
-        let singleton = GnosisSafe::builder(web3).deploy().await.unwrap();
-        let fallback = GnosisSafeCompatibilityFallbackHandler::builder(web3)
-            .deploy()
-            .await
-            .unwrap();
-        let factory = GnosisSafeProxyFactory::builder(web3)
-            .deploy()
-            .await
-            .unwrap();
+    pub async fn new() -> Self {
+        // Using the NODE_HOST here kind of sucks but this is a workaround
+        // for the alloy provider signing issue
```

Comment:
We are an open-source project after all 😄 
```suggestion
        // Using the NODE_HOST here is a compromise but this is a workaround
        // for the alloy provider signing issue
```